### PR TITLE
Leaderboard Issues Handling:

### PR DESCRIPTION
**Pull request by Tthomas63**
Adds a util function `check_bad_scores` which takes a dictionary returned by the method `scores` in CompetitionPhase, and returns a tuple containing: `bad_score_counts`, and `bad_scores`.

- `bad_scores` is a list of bad scores found. They have values of `NaN`.
- `bad_score_counts` is simply a count of the number of bad scores. 

Also adds a check in the `rank_values` method. If a `NaN` score is grabbed, it is set to `0.0` before comparing/performing arithmetic to prevent exceptions. 

Todo: Remove/Reintegrate Exception.

### CODE REVIEW

**Comment by Tthomas63** [comment]
@ckcollab Tests fixed and passing

File: codalab/apps/web/exceptions.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from decimal import Decimal
+
+
+class ScoringException(Exception):
+    value = Decimal('0.0')
+    score_def_id = None
+    result_id = None
+    pass
```
Comment (ckcollab) [inline codalab/apps/web/exceptions.py]: no pass needed?

File: codalab/apps/web/models.py [hunk 2]
```diff
@@ -917,7 +920,12 @@ def rank_values(ids, id_value_pairs, sort_ascending=True, eps=1.0e-12):
         if len(valid_pairs) == 0:
             return {id: 1 for id in ids}
         # Sort and compute ranks
-        sorted_pairs = sorted(valid_pairs.iteritems(), key = operator.itemgetter(1), reverse=not sort_ascending)
+        for k,v in valid_pairs.iteritems():
+            # print("K: {0}; V: {1}".format(k, v))
```
Comment (ckcollab) [inline codalab/apps/web/models.py]: remove cruft

File: codalab/apps/web/templates/web/competitions/_results_page.html [hunk 3]
```diff
@@ -25,6 +25,44 @@
...
+                    <th>
+                        Column Name:
+                    </th>
+                </tr>
+                </thead>
+                <tbody>
+                {% for bad_score in bad_scores %}
+                    <tr>
+                        <td>
+                            {{ bad_score.rnk }}
```
Comment (ckcollab) [inline codalab/apps/web/templates/web/competitions/_results_page.html]: make on one line

**Comment by ckcollab** [comment]
🚢

**Pull request merged**